components/benefit_sponsors/app/controllers/benefit_sponsors/profiles/registrations_controller.rb

```diff
@@ -12,9 +12,11 @@ class RegistrationsController < ::BenefitSponsors::ApplicationController
 
       def new
         @agency = BenefitSponsors::Organizations::OrganizationForms::RegistrationForm.for_new(profile_type: profile_type, portal: params[:portal])
-        authorize @agency
-        authorize @agency, :redirect_home?
-        set_ie_flash_by_announcement unless is_employer_profile?
+        if @agency.present?
+          authorize @agency
+          authorize @agency, :redirect_home?
```

Comment:
I believe we only need this `authorize` line. It also probably shouldn't be conditional. If `@agency` is nil we do want the user to be redirected elsewhere.